src/components/DevelopmentGuide/V4MigrationGuide.mdx

```diff
@@ -0,0 +1,256 @@
+import { Meta, Source } from "@storybook/blocks";
+import { Box, Flex } from "@chakra-ui/react";
+import Banner from "../Banner/Banner";
+import Heading from "../Heading/Heading";
+import Link from "../Link/Link";
+
+<Meta title="Development Guide/v4 Migration Guide" />
+
+# Reservoir v4 Migration Guide
+
+## Table of Contents
+
+- {<Link href="#getting-started">Getting Started</Link>}
+- {<Link href="#breaking-changes">Breaking Changes</Link>}
+- {<Link href="#non-breaking-changes">Non-Breaking Changes</Link>}
+- {<Link href="#v3-support">v3 Support</Link>}
+
+## Back to Foundations
+
+Reservoir v4 includes everything from simple style updates to breaking changes.
+It represents a return to the fundamentals, specifically focusing on the foundational
+layers of HTML, CSS, and JavaScript to make our components more accessible, intuitive,
+and maintainable.
+
+By aligning with web standards, we aim to:
+
+- Simplify the developer's mental model, leading to fewer abstractions and more predictable
+  naming conventions and behaviors.
+- Support accessibility by default through strict adherence to semantic HTML standards.
+- Minimize technical debt and over-engineering by embracing standard HTML and CSS principles
+  and reducing tight coupling with JavaScript logic.
+
+## Getting Started
+
+### Step 1
+
+`npm install @nypl/design-system-react-components@4.0.0`
+
+### Step 2
+
+Run `npx tsc --noEmit` or `npm run lint` to output a list of Typescript errors.
+This list represents the majority of breaking changes you'll need to address.
+
+<Banner
+  content={
+    <>
+      Don't be afraid if this list is long! We promise most of the changes will
+      be quick and easy.
+    </>
+  }
+  my="s"
+  variant="informative"
+/>
+
+### Step 3
+
+Update your code!
+
+## Breaking Changes
+
+### We anticipate the following to be the most common breaking changes.
+
+#### 1. Upgrade your `Template`.
+
+If your application uses the Reservoir `Template`, you’ll need to refactor your code to
```

Comment:
I think we should also call out `TemplateAppContainer` since some apps explicitly use it. Or do you think it's clear enough?

Reply:
I think it is a good idea to include.